src/main/java/edu/harvard/iq/dataverse/DatasetFieldServiceBean.java

```diff
@@ -449,12 +449,24 @@ public JsonObject getExternalVocabularyValue(String termUri) {
      * @param term - the term uri as a string
      */
     public void registerExternalTerm(JsonObject cvocEntry, String term) {
-        String retrievalUri = cvocEntry.getString("retrieval-uri");
-        String prefix = cvocEntry.getString("prefix", null);
         if(term.isBlank()) {
             logger.fine("Ingoring blank term");
             return;
         }
+
+        String cvocUrl = cvocEntry.getString("cvoc-url");
+        String cvocUiUrl = cvocEntry.getString("cvoc-ui-url", null);
+        String retrievalUri = cvocEntry.getString("retrieval-uri");
+        String prefix = cvocEntry.getString("prefix", null);
+        JsonObject headers = cvocEntry.getJsonObject("headers");
+        if (headers == null) {
+            headers = Json.createObjectBuilder().build();
+        }
+
+        if(cvocUiUrl != null) {
+            term = term.replace(cvocUiUrl, cvocUrl);
```

Comment:
Is there a reason the script couldn't use the correct term? It could still get the Ui form for display/front-end query purposes.